### PR TITLE
Update Adventure to 4.11.0 and implement ComponentLogger

### PR DESCRIPTION
**Pull request by JRoy**
:trollface: 

Closes #7873

### CODE REVIEW

File: patches/api/0005-Adventure.patch [hunk 1]
```diff
@@ -487,7 +488,21 @@ index c15861905d7af679c4f6ffc27719a900b0fe4284..d5fdc57ffcfac4eb18d7fbf44ce13257
...
          return server.broadcastMessage(message);
      }
-@@ -1051,6 +1053,19 @@ public final class Bukkit {
+@@ -820,6 +822,13 @@ public final class Bukkit {
+         return server.getLogger();
+     }
+ 
++    // Paper start - Adventure component logger
++    @NotNull
++    public net.kyori.adventure.text.logger.slf4j.ComponentLogger getComponentLogger() {
```
Comment (zml2008) [inline patches/api/0005-Adventure.patch]: A global logger was a mistake in the first place, don't add another one with component support to extend the mistake

File: patches/server/0589-Add-support-for-hex-color-codes-in-console.patch [hunk 2]
```diff
@@ -81,17 +81,12 @@ index 0000000000000000000000000000000000000000..f39ae450e316255d315aeb362bf967ea
 +
 +    private static final String ANSI_RESET = "\u001B[m";
 +
-+    private static final char COLOR_CHAR = 0x7f;
```
Comment (zml2008) [inline patches/server/0589-Add-support-for-hex-color-codes-in-console.patch]: This breaks any error messages containing the literal section symbol

File: patches/server/0008-Adventure.patch [hunk 3]
```diff
@@ -772,6 +772,31 @@ index 0000000000000000000000000000000000000000..2a08e0461db4e699b7e6a1558a4419c8
...
++import org.slf4j.LoggerFactory;
++
++public class ComponentLoggerProviderImpl implements ComponentLoggerProvider {
++    @Override
++    public @NotNull ComponentLogger logger(@NotNull LoggerHelper helper, @NotNull String name) {
++        return helper.delegating(LoggerFactory.getLogger(name), this::serialize);
++    }
++
++    private String serialize(final Component message) {
++        return PaperAdventure.asPlain(message, null);
```
Comment (zml2008) [inline patches/server/0008-Adventure.patch:797]: This shouldn't be asPlain if you want to preserve colors -- it should be something like `HexFormattingConverter.SERIALIZER.serialize(translate(message, Locale.getDefault()))`

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Review by kashike — DISMISSED** [review]
As per @zml2008's comments.

**Review by zml2008 — APPROVED** [review]

File: patches/server/0590-Add-support-for-hex-color-codes-in-console.patch [hunk 4]
```diff
@@ -6,35 +6,66 @@ Subject: [PATCH] Add support for hex color codes in console
...
++import io.papermc.paper.console.HexFormattingConverter;
++import java.util.Locale;
+ import net.kyori.adventure.text.Component;
+ import net.kyori.adventure.text.logger.slf4j.ComponentLogger;
+ import net.kyori.adventure.text.logger.slf4j.ComponentLoggerProvider;
++import net.kyori.adventure.translation.GlobalTranslator;
+ import org.jetbrains.annotations.NotNull;
+ import org.slf4j.LoggerFactory;
+ 
++
```
Comment (jpenilla) [inline patches/server/0590-Add-support-for-hex-color-codes-in-console.patch]: ?

**Review by jpenilla — DISMISSED** [review]

**Review by kennytv — APPROVED** [review]

**Pull request merged**